mvlearn/factorization/ajive.py

```diff
@@ -0,0 +1,921 @@
+    Ph.D.'s pip-installable package, *jive*, the code for which is
+    `linked here <https://github.com/idc9/py_jive>`_.
+
+
+    Examples
+    --------
+    >>> from mvlearn.factorization.ajive import ajive
+    >>> from mvlearn.datasets import load_UCImultifeature
+    >>> Xs, _ = load_UCImultifeature()
+    >>> print(len(Xs)) # number of samples in each view
+    6
+    >>> print(Xs[0].shape, Xs[1].shape) # number of samples in each view
+    (2000, 76) (2000, 216)
+    >>> Ajive = ajive(init_signal_ranks=[2,2])
+    >>> Ajive.fit(Xs)
+    >>> b = Ajive.predict()
+    >>> print(len(b.keys()))
+    6
+    >>> print(b[0]['joint'].shape,b[1]['joint'].shape)
+    (2000, 76) (2000, 216)
+    >>> print(b[0]['individual'].shape,b[1]['individual'].shape)
+    (2000, 76) (2000, 216)
+    >>> print(b[0]['noise'].shape,b[1]['noise'].shape)
+    (2000, 76) (2000, 216)
+
+    References
+    ----------
+    .. [#1ajive] Feng, Qing, et al. “Angle-Based Joint and Individual
+            Variation Explained.” Journal of Multivariate Analysis,
+            vol. 166, 2018, pp. 241–265., doi:10.1016/j.jmva.2018.03.008.
+
+    """
+
+    def __init__(self,
+                 init_signal_ranks,
+                 joint_rank=None, indiv_ranks=None,
+                 center=True,
+                 reconsider_joint_components=True,
+                 wedin_percentile=5, n_wedin_samples=1000,
+                 precomp_wedin_samples=None,
+                 randdir_percentile=95, n_randdir_samples=1000,
+                 precomp_randdir_samples=None,
+                 store_full=True):
+
+        self.init_signal_ranks = init_signal_ranks
+        self.joint_rank = joint_rank
+        self.indiv_ranks = indiv_ranks
+
+        self.center = center
+
+        self.wedin_percentile = wedin_percentile
+        self.n_wedin_samples = n_wedin_samples
+        self.wedin_samples_ = precomp_wedin_samples
+        if precomp_wedin_samples is not None:
+            self.n_wedin_samples = len(list(precomp_wedin_samples.values())[0])
+
+        self.randdir_percentile = randdir_percentile
+        self.n_randdir_samples = n_randdir_samples
+        self.random_sv_samples_ = precomp_randdir_samples
+        if precomp_randdir_samples is not None:
+            self.n_randdir_samples = len(precomp_randdir_samples)
+
+        self.reconsider_joint_components = reconsider_joint_components
+
+        self.store_full = store_full
+
+    def __repr__(self):
+
+        if self.is_fit:
+            r = "joint rank: {}".format(self.common.rank)
+            for bn in self.block_names:
+                indiv_rank = self.blocks[bn].individual.rank
+                r += ", block {} indiv rank: {}".format(bn, indiv_rank)
+            return r
+
+        else:
+            return "No data has been fitted yet"
+
+    def fit(self, blocks, precomp_init_svd=None):
+        r"""
+        Fits the AJIVE decomposition.
+
+        Parameters
+        ----------
+        blocks: dict or list of array-likes
```

Comment:
blocks needs to be renamed to Xs. Also, I don't see a reason why dictionaries should be allowed as input? Just list of array-likes. Find replace should do it